x-pack/plugins/cloud_security_posture/common/scripts/get_tests.js

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Dedicated app for output visualization: https://codesandbox.io/p/sandbox/zen-smoke-vxgs2c
+ * Just copy the generated content of 'cso_test_log.json' into the 'data.json' file in the dedicated app
+ * */
+
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+
+// ==============================
+//           Setup
+// ==============================
+
+// Directories to iterate over
+const FTR_SERVERLESS =
+  'x-pack/test_serverless/functional/test_suites/security/ftr/cloud_security_posture';
+const FTR_API_INTEGRATION = 'x-pack/test/api_integration/apis/cloud_security_posture';
+const FTR_CSP_API = 'x-pack/test/cloud_security_posture_api';
+const FTR_CSP_FUNCTIONAL = 'x-pack/test/cloud_security_posture_functional';
+const UNIT_TEST_CSP = 'x-pack/plugins/cloud_security_posture';
+
+const directoryPaths = [
+  FTR_SERVERLESS,
+  FTR_API_INTEGRATION,
+  FTR_CSP_API,
+  FTR_CSP_FUNCTIONAL,
+  UNIT_TEST_CSP,
+];
+
+// Output directories and file paths
+const MD_FILE_PATH = path.join(
+  'x-pack/plugins/cloud_security_posture/common/dev_docs',
+  '__auto_generated_csp_requirements_test_coverage.md'
+);
+const CSP_TEST_LOGS_FILE_PATH = path.join(__dirname, '__auto_generated_csp_test_log.json');
+
+// Test output data
+const testsLogOutput = [];
+const regex = /\b(?:describe\.skip|describe|it\.skip|it)\(['`]/;
```

Comment:
For serverless there is also important information can be taken from `this.tags()` in the test file. There are 2 tags we are interested in `skipMKI` and `failsOnMKI`